Sources/StreamVideoSwiftUI/CallContainer.swift

```diff
@@ -108,22 +108,25 @@ public struct WaitingLocalUserView<Factory: ViewFactory>: View {
     
     public var body: some View {
         ZStack {
-            if let localParticipant = viewModel.localParticipant {
-                LocalVideoView(
-                    viewFactory: viewFactory,
-                    participant: localParticipant,
-                    idSuffix: "waiting",
-                    callSettings: viewModel.callSettings,
-                    call: viewModel.call
-                )
-            } else {
-                DefaultBackgroundGradient()
-            }
-            
-            VStack {
-                Spacer()
-                viewFactory.makeCallControlsView(viewModel: viewModel)
-                    .opacity(viewModel.callingState == .reconnecting ? 0 : 1)
+            DefaultBackgroundGradient()
+                .edgesIgnoringSafeArea(.all)
+
+            VStack() {
+                if let localParticipant = viewModel.localParticipant {
+                    LocalVideoView(
+                        viewFactory: viewFactory,
+                        participant: localParticipant,
+                        idSuffix: "waiting",
+                        callSettings: viewModel.callSettings,
+                        call: viewModel.call
+                    )
+                } else {
+                    Spacer()
+                }
+
+                if viewModel.callingState != .reconnecting {
+                    viewFactory.makeCallControlsView(viewModel: viewModel)
```

Comment:
The recommended SwiftUI way is to use opacity instead of conditions for hiding the views. Any particular reason we changed this?

Reply:
It was mentioned few times in WWDC videos. In this case, when you do an if, the view will loose its identity on reconnect, which will trigger additional computations when reconnection finishes. Additionally, reconnect doesn't happen often, so this view is visible almost always.

Reply:
Hmm I see what you mean. However, even with opacity and hidden there differences in the first one the view is rendered but in the second it isn't (but still takes place in the layout). Can we change it to hidden?

Reply:
Thanks for the clarifications :) I have reverted it